lib/autoupdate/tools/utils.go

```diff
@@ -194,6 +194,11 @@ type packageURL struct {
 
 // teleportPackageURLs returns URLs for the Teleport archives to download.
 func teleportPackageURLs(ctx context.Context, uriTmpl string, baseURL, version string) ([]packageURL, error) {
+	semVersion, err := semver.NewVersion(version)
```

Comment:
nit: should we be only dealing in `semver.Version` throughout rather than in strings that we have to fallibly parse?